### DISCUSSION TITLE
CGAL post classification optimization

### DISCUSSION THREAD

**Discussion by pmff96** [CGAL/cgal#5505] [Q&A] [answered]
Good afternoon, I apologize for another inconvenience, but I had a question regarding the CGAL classification package, or the post processing rather. What I would like to know is that if CGAL contains any tools/packages for post classification optimization? (e.g. correcting misclassified points). The example dataset I used contained power lines, transmission towers and vegetation, like the dataset portrayed in the classification demo tutorial video. I did have some misclassified points, like some line points being classified as vegetation. I would like to optimize this classification results and was looking for the tools to do it (without changing the ground truth and reclassifying again). Does CGAL provide support to this post classification processing? Once again, I apologize for the inconvenience, I'm fairly new to CGAL and point cloud processing in general.

**Comment by sgiraudot** [comment, marked as answer]
There's no inconvenience, really, this place is for asking questions, so you are very welcome to do so!

So far there is no post processing tool for Classification. One way to have better regularized results is to use `classify_with_local_smoothing()` or `classify_with_graphcut()` (instead of  `classify()`) which can help getting rid of small errors, but maybe you have already looked into that. Both methods have parameters that you can tune to activate more or less regularization, but of course this requires some trial-and-error steps to get the best parameters for your use case.

If you just want to do some manual "clean-up", then you can use the demo to select some points and add them to the training set (which will also change their label as if they had been classified like this), but I imagine you don't want to do that as you don't want to change the ground truth.

**Reply by pmff96** [reply]
Yes exactly, I was trying to do post classification processing without manually improving the ground truth data and reclassifying it, however, if I don't find another way to do it I might go for it, thank you for your help!
